core/src/main/scala/kafka/server/AlterPartitionManager.scala

```diff
@@ -91,7 +91,8 @@ object AlterPartitionManager {
       config = config,
       channelName = "alterPartition",
       threadNamePrefix = threadNamePrefix,
-      retryTimeoutMs = Long.MaxValue
+      networkClientRetryTimeoutMs = if (config.processRoles.isEmpty) config.controllerSocketTimeoutMs else config.brokerSessionTimeoutMs / 2,
```

Comment:
we don't want to change this, right?

Reply:
> let's leave this one alone for now and file a follow-up JIRA for it

I've opened https://issues.apache.org/jira/browse/KAFKA-14394 to track this.

Reply:
I believe https://issues.apache.org/jira/browse/KAFKA-14394 can now be closed.  We do want infinite retries on the broker side for AlterPartition requests as per https://github.com/apache/kafka/pull/9564#discussion_r527326231.